repos/system_upgrade/el7toel8/actors/cloud/checkhybridimage/libraries/checkhybridimage.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledRPM, HybridImageDetected, FirmwareFacts
+from leapp import reporting
+from leapp.libraries.common import rhui
+
+
+BIOS_PATH = '/boot/grub2/grubenv'
+EFI_REL_PATH = '../efi/EFI/redhat/grubenv'
+
+
+def is_grubenv_symlink_to_efi():
+    """
+    Check whether '/boot/grub2/grubenv' is a relative symlink to
+    '/boot/efi/EFI/redhat/grubenv'.
+    """
+    return os.path.islink(BIOS_PATH) and os.readlink(BIOS_PATH) == EFI_REL_PATH
```

Comment:
Yes, in case of the official Azure image it will always be like that. There are kickstarts for this. IMHO we can't predict what users will do. I think they can do literally everything :)

Reply:
Implemented as discussed offline.